### PR TITLE
Metrics: Add IPAM actions metrics.

### PR DESCRIPTION
**Pull request by eloycoto**
Add a new metric related to the IPAM operations in Cilium.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5692)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Review by jrajahalme — APPROVED** [review]
LGTM

File: Documentation/configuration/metrics.rst [hunk 1]
```diff
@@ -118,6 +118,11 @@ Kubernetes
 * ``kubernetes_events_total``: Number of Kubernetes events received labeled by
   scope, action and the execution result
 
+Ipam
```
Comment (ianvernon) [inline Documentation/configuration/metrics.rst]: Ipam --> IPAM

File: Documentation/configuration/metrics.rst [hunk 2]
```diff
@@ -118,6 +118,11 @@ Kubernetes
...
+------
+
+* ``ipam_events_total``: Number of ipam events received labeled by action and
```
Comment (ianvernon) [inline Documentation/configuration/metrics.rst]: ipam --> IPAM

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -416,6 +416,16 @@ var (
...
 	}, []string{LabelScope, LabelAction, LabelStatus})
+
+	// Ipam events
+
+	// IpamEvent is the number of ipam events received labeled by action and
+	// datapath family type
+	IpamEvent = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: Namespace,
+		Name:      "ipam_events_total",
+		Help:      "Number of ipam events received labeled by action and datapath family type",
```
Comment (ianvernon) [inline pkg/metrics/metrics.go]: ipam --> IPAM

**Review by ianvernon — COMMENTED** [review]
LGTM apart from capitalization in documentation for IPAM.

**Review by rlenglet — APPROVED** [review]
LGTM.

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by tgraf** [comment]
This was labeled for the 1.4 milestone and is clearly a feature. Why are we merging this?

**Comment by ianvernon** [comment]
I did not notice the milestone :/ and it was marked as `pending-review` - sorry about that. Would you like me to revert @tgraf ? Happy to take that on since I am the one who merged.

**Comment by tgraf** [comment]
@ianvernon I had a look and it seems low risk so I'm fine to keep it

**Pull request merged**